### ISSUE TITLE
Ability in edge proxy mode to force https (ignore/missing X-forwarded-proto)

### ISSUE THREAD

**Issue by michaelpearce-gain** [keycloak/keycloak#12778]
### Describe the bug

Previously you could provide the full uri for keycloak, meaning if behind a few Loadbalancers it did not matter, and the ui would correctly use the uri for all parts.

In the new quarkus version it seems we have to use "edge" proxy, but it seems its heavily dependent on -" X-Forwarded-Proto" which unfortunately in our env is being lost in the multiple Loadbalancers as is coming in as http even so url should be https as SSL is offloaded up the stack.

Can we ask that theres a way to FORCE quarkus version to use https regardless of  X-Forwarded-Proto so we can migrate up from wildfly.

### Version

18.0.1

### Expected behavior

ability to for https and ignore  X-Forwarded-Proto

### Actual behavior

currently it uses  X-Forwarded-Proto and no way to override

### How to Reproduce?

have a loadbalancer that does not set  X-Forwarded-Proto

### Anything else?

no

**Comment by andre-nascimento-rh** [comment]
Hello, @michaelpearce-gain ,

Thanks for reaching out and sorry you are facing issues when migrating to newer Keycloak versions.

However, this issue seems like just a usability issue, or an RFE (Request For Enhancement), then it is not a bug in our point of view. Therefore, we're gonna moving this issue to "Backlog" to be handled some moment after the version 22 release.

If it is possible, enrich the issue description by providing an example of one LoadBalancer that does not set "X-Forwarded-Proto" header.

Thanks in advance.

**Comment by michaelpearce-gain** [comment]
I disagree this isn't a new RFE / usability issue, This surely is a regression. If it supported in 18.x but broken not there in 20.x.

If needed i can raise a formal RH ticket as a customer. But this will block our upgrade of RH SSO in future.

**Comment by michaelpearce-gain** [comment]
Setup is a Client -> F5 loadbalancer (L4) -> Nginx (L7) -> App , its lost between

**Comment by vmuzikar** [comment]
@michaelpearce-gain Hello, thanks for the report. I'm sorry you are facing issues with the proxy settings. Could you please share your Keycloak settings to help us reproduce the issue? In particular, did you try setting `hostname-url` with https scheme?

**Comment by vmuzikar** [comment]
I'm going to close this. Feel free to reopen whenever you are able to provide more details. Thanks.
